### ISSUE TITLE
TFException on Multinomial when launching the Graph Model

### ISSUE THREAD

**Issue by Dastyn** [Unity-Technologies/ml-agents#306]
Hi there!

This issue looks like issue #**146**, but this time is about Multinomial.
It appears when launching the GraphModel on a trained game of my own.

This is the exception:

TFException: NodeDef mentions attr 'output_dtype' not in Op<name=Multinomial; signature=logits:T, num_samples:int32 -> output:int64; attr=seed:int,default=0; attr=seed2:int,default=0; attr=T:type,allowed=[DT_FLOAT, DT_DOUBLE, DT_INT32, DT_INT64, DT_UINT8, DT_INT16, DT_INT8, DT_UINT16, DT_HALF]; is_stateful=true>; NodeDef: multinomial/Multinomial = Multinomial[T=DT_FLOAT, output_dtype=DT_INT64, seed=0, seed2=0](dense_3/MatMul, multinomial/Multinomial/num_samples). (Check whether your GraphDef-interpreting binary is up to date with your GraphDef-generating binary.).
TensorFlow.TFStatus.CheckMaybeRaise (TensorFlow.TFStatus incomingStatus, System.Boolean last) (at <6ed6db22f8874deba74ffe3e566039be>:0)
TensorFlow.TFGraph.Import (TensorFlow.TFBuffer graphDef, TensorFlow.TFImportGraphDefOptions options, TensorFlow.TFStatus status) (at <6ed6db22f8874deba74ffe3e566039be>:0)
TensorFlow.TFGraph.Import (System.Byte[] buffer, TensorFlow.TFImportGraphDefOptions options, TensorFlow.TFStatus status) (at <6ed6db22f8874deba74ffe3e566039be>:0)
TensorFlow.TFGraph.Import (System.Byte[] buffer, System.String prefix, TensorFlow.TFStatus status) (at <6ed6db22f8874deba74ffe3e566039be>:0)
CoreBrainInternal.InitializeCoreBrain () (at Assets/ML-Agents/Scripts/CoreBrainInternal.cs:112)
Brain.InitializeBrain () (at Assets/ML-Agents/Scripts/Brain.cs:187)
Academy.Awake () (at Assets/ML-Agents/Scripts/Academy.cs:124)

Some more info on the environment:
- Unity 2017.3.0f3
- TFSharpPlugin downloaded from https://s3.amazonaws.com/unity-agents/TFSharpPlugin.unitypackage
- Windows 10
- tensorflow-gpu (1.5.0)

Any idea, hint, help?

BR,
Dastyn

**Comment by Dastyn** [comment]
Some more info on the environment:
- Unity 2017.3.0f3
- TFSharpPlugin downloaded from https://s3.amazonaws.com/unity-agents/TFSharpPlugin.unitypackage
  (but the same exception was raised with version 0.2 of the same plugin)
- Windows 10
- tensorflow-gpu (1.5.0)

**Comment by vincentpierre** [comment]
Can you try with tensorflow r1.4? It could be that the tensorflowsharp package is incompatible.

**Comment by Dastyn** [comment]
You are perfectly right!
Downgrading to tensorflow v1.4 fixed the issue.
Thanks a lot for your quick help.
Dastyn

**Comment by potatoorgylt** [comment]
That's odd. I get exactly the same issue. Downgraded to 1.4. It's still the same. Any suggestions?

**Comment by Dastyn** [comment]
Check if, once you switched to tensorflow python environment (ie: activate tensorflow under your command prompt), you get the right tensorflow version: pip list (or pip3 list).

**Comment by ErikSvedin** [comment]
I'm experiencing this same error as well.

pip freeze gives me `tensorflow==1.4.0` and I've restarted Unity just to be sure.

**Comment by ematsuno** [comment]
https://s3.amazonaws.com/unity-agents/TFSharpPlugin.unitypackage

is giving me 

<Error>
<Code>AllAccessDisabled</Code>
<Message>All access to this object has been disabled</Message>
<RequestId>20ADEDDFC5D23DD7</RequestId>
<HostId>
tS1QaFniGXHDhYUpFWgp0vZjZ3pVXci9EM7ZUffZJMPpNCOCPkHknc47zK82PTWYaf+z184FOc0=
</HostId>
</Error>

**Comment by awjuliani** [comment]
Hi @ematsuno,

That is an old link to a previous version of the TF# plugin. Here is the most current link for the v0.5 release: https://s3.amazonaws.com/unity-ml-agents/0.5/TFSharpPlugin.unitypackage

**Comment by kibotu** [comment]
is there are any way to see the latest version? there are a lot of tutorials using different versions at this point and the s3 is not helpful figuring out if you're up-to-date ^^'

**Comment by husnainyousaf88** [comment]
DT_UINT8, DT_INT16, DT_INT8, DT_UINT16, DT_HALF]; is_stateful=true>; NodeDef: multinomial_3/Multinomial = Multinomial[T=DT_FLOAT, output_dtype=DT_INT64, seed=6006, seed2=233](Log_3, multinomial_3/Multinomial/num_samples). (Check whether your GraphDef-interpreting binary is up to date with your GraphDef-generating binary.).
TensorFlow.TFStatus.CheckMaybeRaise (TensorFlow.TFStatus incomingStatus, System.Boolean last) (at <1fe2de69842a4a4ba15256b83cca05f3>:0)
TensorFlow.TFGraph.Import (TensorFlow.TFBuffer graphDef, TensorFlow.TFImportGraphDefOptions options, TensorFlow.TFStatus status) (at <1fe2de69842a4a4ba15256b83cca05f3>:0)
TensorFlow.TFGraph.Import (System.Byte[] buffer, TensorFlow.TFImportGraphDefOptions options, TensorFlow.TFStatus status) (at <1fe2de69842a4a4ba15256b83cca05f3>:0)
TensorFlow.TFGraph.Import (System.Byte[] buffer, System.String prefix, TensorFlow.TFStatus status) (at <1fe2de69842a4a4ba15256b83cca05f3>:0)
MLAgents.CoreBrainInternal.InitializeCoreBrain (MLAgents.Batcher brainBatcher) (at Assets/ML-Agents/Scripts/CoreBrainInternal.cs:133)
MLAgents.Brain.InitializeBrain (MLAgents.Academy aca, MLAgents.Batcher brainBatcher) (at Assets/ML-Agents/Scripts/Brain.cs:207)
MLAgents.Academy.InitializeEnvironment () (at Assets/ML-Agents/Scripts/Academy.cs:288)
MLAgents.Academy.Awake () (at Assets/ML-Agents/Scripts/Academy.cs:227)

having the issue with mlagents 0.5 and tfsharp 0.5 any solution?
